### DISCUSSION TITLE
你们域名分流真的生效吗？

### DISCUSSION THREAD

**Discussion by lddsb** [joyanhui/ikuai-bypass#37] [公告]
我设置了域名分流强制走adsl1，但是还是走到其他出口去了

traceroute截图：
![image](https://github.com/user-attachments/assets/0f79f5d4-111c-4080-9ba6-c681b93b813b)

域名分流截图： 
![image](https://github.com/user-attachments/assets/0556e305-12a8-49ee-88db-9c008ead448c)

**Comment by mmd1989** [comment]
Ikuai的域名分流，官方提示是要跟多线路DNS一起用，我看教程里面没有这块

**Comment by joyanhui** [comment]
域名分流必然是有效的 而且比较好用 

ikuai-bypass 支持两种分流模式的规则自动更新。

域名分流 配合 虚拟wan口的方式 实际效果要比 端口转发的模式更灵活更容易一些。

简单的说  域名分流  通常是配合多wan口（可以是虚拟网口），我们只需要让其中一个wan口的上游是openwrt，然后openwrt的上游是爱快的lan 并在爱强制openwrt的流量出口到 正常的那个wan口即可。  其他和官网的域名分流教程一样的。

爱快是国内公司，官网自然不可能给你提供手把手的教程，毕竟你要干的那个事情官网能公开说嘛? 

在 https://github.com/joyanhui/ikuai-bypass/blob/main/README.md  有详细描述，也有top图，也有引用的恩山的几个大佬的帖子

如果你的没有生效，还是自己检查一下配置。

**Comment by lddsb** [comment]
解决了，不生效是因为我多路DNS里面，wan1口的DNS设置为了运营商的DNS，没想到居然有坑，哎，改成自建的MosDNS地址就好了

![image](https://github.com/user-attachments/assets/c2358d1d-9061-4c9f-9805-ba949ffaf114)

重新traceroute
![image](https://github.com/user-attachments/assets/cf0b846d-a880-4fc5-a9c2-da511406f058)

**Reply by refined-fish** [reply]
> 解决了，不生效是因为我多路DNS里面，wan1口的DNS设置为了运营商的DNS，没想到居然有坑，哎，改成自建的MosDNS地址就好了
> 
> ![image](https://private-user-images.githubusercontent.com/20765867/387231994-c2358d1d-9061-4c9f-9805-ba949ffaf114.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MTY5NjUsIm5iZiI6MTczMzkxNjY2NSwicGF0aCI6Ii8yMDc2NTg2Ny8zODcyMzE5OTQtYzIzNThkMWQtOTA2MS00YzlmLTk4MDUtYmE5NDlmZmFmMTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDExMzEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkNTkwNmZmZWZiZDk5NzZjYTg3NDkxMjVjYTFjNjJhZmZkNzc3MjBjM2VkMjNkMmViYjgxODI0NmRkMWVkNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ETstO_t2SEpripDWwlxBiXgnCUCD3js3XuSxPFM9OEc)
> 
> 重新traceroute ![image](https://private-user-images.githubusercontent.com/20765867/387232325-cf0b846d-a880-4fc5-a9c2-da511406f058.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MTY5NjUsIm5iZiI6MTczMzkxNjY2NSwicGF0aCI6Ii8yMDc2NTg2Ny8zODcyMzIzMjUtY2YwYjg0NmQtYTg4MC00ZmM1LWE5YzItZGE1MTE0MDZmMDU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDExMzEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYyYmU3ODczODk0OWZmYmYxYzYxZjQ3OTVmNDFlMDUwNTMwNmNkMTI0ZGVmZjZiMTEyN2RhOTY3NGJkMDgyY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6fmiI-GLjV458QyiXSUBpKOl44oSF-aL7keAYuRgQBw)

你两个线路都设置成一样的DNS，那还不如不使用多线DNS的功能，多线DNS的意义在于，走不同线路出去的流量都能找到最佳的DNS返回的最佳IP，简单讲就是为了分配不同的DNS用的，你都用一样，可以直接关闭该功能，直接用爱快的DNS代理功能
